civic-tech-club/src/components/BudgetVis.js

```diff
@@ -0,0 +1,226 @@
+import React, { useState, useEffect, useRef } from 'react';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import zoomPlugin from 'chartjs-plugin-zoom'; // Import the zoom plugin
+import data from '../data/budget.json';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+// Register the required Chart.js components and the zoom plugin
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler,
+  zoomPlugin // Register the zoom plugin here
+);
+
+// Helper to generate unique colors
+const COLORS = [
+  'rgba(75, 192, 192, 1)',
+  'rgba(255, 99, 132, 1)',
+  'rgba(54, 162, 235, 1)',
+  'rgba(255, 206, 86, 1)',
+  'rgba(153, 102, 255, 1)',
+  'rgba(255, 159, 64, 1)',
+];
+
+const BudgetVis = () => {
+  const [chartTitle, setChartTitle] = useState('');
+  const [chartData, setChartData] = useState({ labels: [], datasets: [] });
+  const chartRef = useRef(null); // Create a ref for the chart
+
+  useEffect(() => {
+    // This code will run once when the component loads
+    handleChartChange('Gross Operating Expenditures');
+  }, []); // The empty array makes it run only once, when the component mounts
+
+  const handleChartChange = (title) => {
+    setChartTitle(title); // Set chart title for debugging or displaying
+    const newChartData = getLine(title, data); // Pass title directly to getLine function
+    setChartData(newChartData); // Update chart data with the new dataset
+    resetZoom();
+  };
+
+  // Generate the line chart data with custom colors
+  function getLine(budgetIndex, data) {
+    const chartdata = {
+      labels: ['2024', '2025', '2026', '2027'], // Preset labels for the X-axis (years)
```

Comment:
We might be able to change this in the future. I think the best way to enable this is to modify our python script to put the year values for each budget summary into an array, so that we can iterate over it. However, the year data isn't going to change until probably 2027, so I think it's okay for now.

Reply:
ah I see. Perfect.